### PR TITLE
Issue/13328 restore progress

### PR DESCRIPTION
**Pull request by zwarm**
Parent #13328

This PR implements the "progress" wizard step/view for the restore request and is mostly a clone of Backup Download progress with a few temporary navigations until the last step in the wizard has been implemented. 

**Merge Instructions**
- Make sure #13773 & #13775 have been merged (sorry, but they need to build upon each other)
- Remove Not Ready For Merge Label
- Merge as normal

**Cloned Classes**
RestoreProgressAdapter
RestoreProgressFragment
RestoreProgressStateListItemBuilder

**Notes**:
- Ignore styling - this is being handled separately 
- Ignore unit test placement in packages - this is being handled separately 
- RestoreActivity & RestoreViewModel (+unit tests) will continue to get updated as I move throughout the steps of the wizard

**To test:**
**Warning**: The Restore site button is hooked up, so if you press it your site will be backed up!
- Launch the app with RestoreFeatureConfig flag on and the BackupFeatureFlag on
- Navigate to ActivityLog
- Tap the more menu
- Select the restore option
- Tap the Restore Site button on the Details View
- Tap the Confirm restore site on the Warning View
Verify
- (1) The progress view is shown
- (2) If there is an error, the wizard will close (eventually it will go to the complete view)
- (3) When the restore finishes, the wizard will close (eventually it will go to the complete view)
- (4) Tapping the OK, notify me button, Back Button and X will close the wizard

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModel.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModel.kt:157]: Minor (🔍): Consider adding test for this and any other suchlike public functions.

PS: The same apply for the `RestoreViewModel`, its `onProgressExit()` function and its other public functions.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt:20]: Warning (⚠️): Consider adding tests for this use case.

PS: Also the same applies to the `GetBackupDownloadStatusUseCase` use case, which doesn't have any as well.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt]: 1) Question (❓): Why 10'' seconds?
2) Question (❓): There is another delay at the end of this function, do we need both of them?

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt:24]: Praise (❤️): I am loving this `Flow` solution, kudos!

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt]: Minor (🔍): Consider removing `statusForSite` parameter since it is not used anywhere. What I mean is that these two lines can be merge: `val rewind = activityLogStore.getRewindStatusForSite(site)?.rewind`

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/viewholders/RestoreAdditionalInformationViewHolder.kt
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/viewholders/RestoreAdditionalInformationViewHolder.kt:14]: Warning (⚠️): This `Restore` related view holder is using a `Backup` related layout file. This is most probably an oversight, right?

PS: The same applies for the `RestoreProgressViewHolder` class.

File: WordPress/src/main/res/layout/restore_progress_fragment.xml
Comment (ParaskP7) [inline WordPress/src/main/res/layout/restore_progress_fragment.xml:6]: Minor (🔍): This comment does not relate to this layout file but its corresponding `Backup` layout file, where there is an unused `xmlns:tools` namespace. Consider cleaning this up.

**Review by ParaskP7 — COMMENTED** [review]
👋 @zwarm  !

I have reviewed and tested this PR as per the instruction, great job! Also, kudos for the `Flow` usage on the use case! 🌟 ❤️  

I have left some comments (🔍 + ⚠️) and few questions (❓), which I would like to discuss before approving this PR.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt]: @ParaskP7 - While testing I discovered there is indeed another state called "queued" which is not accounted for in the existing services. It is not in FluxC either. Once that state is added to fluxc then I can properly handle it in the use case. Looking back at the existing service, they use a "now" parameter to either kick off the status check immediately OR delay for 10 seconds. Since this use case is always run after submittal of the restore request (again mimicking the existing), I just use the delay. I plan on updating FluxC when I update getting progress file names. 

What I will do is move the delay to after the network check - it will have the same effect. in commit 4bd7d42

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModel.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModel.kt:157]: Noted and added the task. 👍

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt:20]: Noted and added the task. 👍

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/viewholders/RestoreAdditionalInformationViewHolder.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/viewholders/RestoreAdditionalInformationViewHolder.kt:14]: That is correct - "backup" is generic - I will be renaming to backup_restore later today :)

File: WordPress/src/main/res/layout/restore_progress_fragment.xml
Comment (zwarm) [reply WordPress/src/main/res/layout/restore_progress_fragment.xml:6]: Noted - there will only be 1 fragment xml in the future - they are all the same.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt]: In commit c5b9f6d

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt]: Thanks for the detailed explanation @zwarm ! 🌟 

I am guessing the above is a response to `Question.1`. How about `Question.2`?

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt:20]: Thanks @zwarm ! Just to make it clear, this will be done on another PR, right?

**Comment by zwarm** [comment]
@ParaskP7 - Let me know what else needs to be done before you give you approval? Thanks.

**Comment by ParaskP7** [comment]
> @ParaskP7 - Let me know what else needs to be done before you give you approval? Thanks.

👋 @zwarm !

I have two pending questions for you, then I'll approve this PR. Can you check the comments I haven't yet resolved?

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt]: I removed the second delay in that same commit - moved it so that it can be reused.

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (zwarm) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt:20]: Yes, use case unit tests will be in a separate PR - not yet started

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/usecases/GetRestoreStatusUseCase.kt]: Oh sorry, missed that, you are right! 🙏

**Review by ParaskP7 — APPROVED** [review]

**Comment by zwarm** [comment]
Many thanks @ParaskP7

**Pull request merged**
